runtime/resolver.go

```diff
@@ -40,10 +41,18 @@ type Resolver interface {
 type ResolverResult struct {
 	// Data is a JSON encoded array of objects.
 	Data []byte
+	// Schema is the schema for the Data
+	Schema *runtimev1.StructType
 	// Cache indicates whether the result can be cached.
 	Cache bool
 }
 
+// CachedResolverResult is subset of ResolverResult that is cached
+type CachedResolverResult struct {
```

Comment:
Would suggest calling this `ResolveResult` because it is a result returned from `Resolve` (note: not "resolve", not "resolver"). And then move the definition to after `the `ResolveOptions` struct.

"Cached" is an implementation detail, and it will actually not always have been served from cache :)